testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveSparkCompatibility.java

```diff
@@ -423,6 +424,78 @@ private void testReadTrinoCreatedTable(String tableName, String tableFormat)
         onTrino().executeQuery("DROP TABLE " + trinoTableName);
     }
 
+    @Test(groups = {HIVE_SPARK, PROFILE_SPECIFIC_TESTS})
+    public void testReadSparkInvalidDatePartitionName()
+    {
+        String sparkTableName = "test_trino_reading_spark_date_type_partitioned_" + randomTableSuffix();
+        String trinoTableName = format("%s.default.%s", TRINO_CATALOG, sparkTableName);
+
+        onSpark().executeQuery(format("CREATE TABLE default.%s (value integer) PARTITIONED BY (dt date)", sparkTableName));
+
+        // Spark allows creating partition with invalid date format
+        // Hive deny creating such partitions, but allows reading
+        onSpark().executeQuery(format("INSERT INTO %s PARTITION(dt='1965-09-10 23:59:59.999999999') VALUES (1)", sparkTableName));
+        onSpark().executeQuery(format("INSERT INTO %s PARTITION(dt='1965-09-10 23:59:59') VALUES (2)", sparkTableName));
```

Comment:
I would rather have this tested against Hive, as it's the reference implementation.
I don't see a need to test this with Spark, but we can leave it if you want.